webapp/src/app/participant-view/participant-stage-view/exp-leader-reveal/exp-leader-reveal.component.ts

```diff
@@ -1,80 +1,66 @@
-import { Component, Inject, signal, Signal } from '@angular/core';
+import { Component, Inject, Input, signal, Signal } from '@angular/core';
+import { MatButtonModule } from '@angular/material/button';
+import { injectQueryClient } from '@tanstack/angular-query-experimental';
 
 import { ProviderService } from 'src/app/services/provider.service';
+import { updateLeaderRevealStageMutation } from 'src/lib/api/mutations';
 import { Participant } from 'src/lib/participant';
 import { PARTICIPANT_PROVIDER_TOKEN } from 'src/lib/provider-tokens';
-import { StageKind } from 'src/lib/types/stages.types';
+import { ExpStageVoteReveal } from 'src/lib/types/stages.types';
 import { VoteReveal } from 'src/lib/types/votes.types';
 
 @Component({
   selector: 'app-exp-leader-reveal',
   standalone: true,
-  imports: [],
+  imports: [MatButtonModule],
   templateUrl: './exp-leader-reveal.component.html',
   styleUrl: './exp-leader-reveal.component.scss',
 })
 export class ExpLeaderRevealComponent {
+  // Reload the internal logic dynamically when the stage changes
+  @Input({ required: true })
+  set stage(value: ExpStageVoteReveal) {
+    this._stage = value;
+
+    this.stageData = this.stage.config;
+    this.everyoneReachedThisStage = this.participant.everyoneReachedCurrentStage(this.stage.name);
+  }
+
+  private queryClient = injectQueryClient();
```

Comment:
Needs a comment to say what this is I think.

Reply:
... I suspect this is not the right design... I think you should be simply be doing a firebase groupQuery for the participants, and then when they are all at done at this stage, you move to the next...